node/src/components/linear_chain_sync/operations.rs

```diff
@@ -493,12 +497,11 @@ async fn fast_sync_to_most_recent(
             // If we could not fetch, we can stop when the most recent header:
             // 1. has our protocol version
             // 2. is in the current era
-            None if most_recent_block_header.protocol_version() == current_version
-                && is_current_era(
-                    &most_recent_block_header,
-                    &trusted_key_block_info,
-                    chainspec,
-                ) =>
+            None if is_current_era(
```

Comment:
I see the check for `protocol_version` is now missing. If that's expected, please update the comment above the `if` statement.